@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -109,11 +109,18 @@ export class VersionedLayerClient {
                 const quadTreeIndex = await this.getPartitions(
                     quadKeyPartitionsRequest
                 ).catch(error => Promise.reject(error));
-                return this.downloadPartition(
-                    quadTreeIndex.subQuads[0].dataHandle,
-                    abortSignal,
-                    dataRequest.getBillingTag()
-                );
+
+                if (quadTreeIndex.status && quadTreeIndex.status === 400) {
+                    return Promise.reject(quadTreeIndex);
+                }
+
+                return quadTreeIndex.subQuads
```

Comment:
what the quadTreeIndex will be if getPartitions will reject error?

Reply:
It will be rejected in the line #123.